trollflow2/plugins/__init__.py

```diff
@@ -128,32 +128,46 @@ def resample(job):
                                      conf)
         LOG.debug('Resampling to %s', str(area))
         if area == 'None':
-            minarea = get_config_value(product_list,
-                                       '/product_list/areas/' + str(area),
-                                       'use_min_area')
-            maxarea = get_config_value(product_list,
-                                       '/product_list/areas/' + str(area),
-                                       'use_max_area')
-            native = conf.get('resampler') == 'native'
-            if minarea is True:
-                job['resampled_scenes'][area] = scn.resample(scn.min_area(),
-                                                             **area_conf)
-            elif maxarea is True:
-                job['resampled_scenes'][area] = scn.resample(scn.max_area(),
-                                                             **area_conf)
-            elif native:
-                job['resampled_scenes'][area] = scn.resample(resampler='native')
-            else:
-                # The composites need to be created for the saving to work
-                if not set(scn.keys()).issuperset(scn.wishlist):
-                    LOG.debug("Generating composites for 'null' area (satellite projection).")
-                    scn.load(scn.wishlist, generate=True)
-                job['resampled_scenes'][area] = scn
+            job['resampled_scenes'][area] = _resample_null_area(product_list, scn, conf, area_conf)
         else:
             LOG.debug("area: %s, area_conf: %s", area, str(area_conf))
             job['resampled_scenes'][area] = scn.resample(area, **area_conf)
 
 
+def _resample_null_area(product_list, scn, conf, area_conf):
+    use_coarsest_area = _get_native_area(product_list, 'use_coarsest_area', legacy='use_min_area')
+    use_finest_area = _get_native_area(product_list, 'use_finest_area', legacy='use_max_area')
+    native = conf.get('resampler') == 'native'
+    if use_coarsest_area is True:
+        return scn.resample(scn.coarsest_area(), **area_conf)
+    if use_finest_area is True:
+        return scn.resample(scn.finest_area(), **area_conf)
+    if native:
+        return scn.resample(resampler='native')
+
+    # The composites need to be created for the saving to work
+    if not set(scn.keys()).issuperset(scn.wishlist):
+        LOG.debug("Generating composites for 'null' area (satellite projection).")
+        scn.load(scn.wishlist, generate=True)
+    return scn
+
+
+def _get_native_area(product_list, area_name, legacy='use_min_area'):
```

Comment:
I'm not convince `area_name` is the right parameter name here... would you mind changing it to, say, `area_grouping_method`?
Same for `legacy`, maybe `legacy_grouping_method`?